bin/stress-test/src/main.rs

```diff
@@ -0,0 +1,377 @@
+        Felt::default(),
+        rng,
+    )
+    .expect("note creation failed")
+}
+
+/// Create a new account with a given public key and anchor block. Generates the seed from the given
+/// index.
+fn create_account(anchor_block: &BlockHeader, public_key: PublicKey, index: u64) -> AccountId {
+    let init_seed: Vec<_> = index.to_be_bytes().into_iter().chain([0u8; 24]).collect();
+    let (new_account, _) = AccountBuilder::new(init_seed.try_into().unwrap())
+        .anchor(anchor_block.try_into().unwrap())
+        .account_type(AccountType::RegularAccountImmutableCode)
+        .storage_mode(AccountStorageMode::Private)
+        .with_component(RpoFalcon512::new(public_key))
+        .with_component(BasicWallet)
+        .build()
+        .unwrap();
+    new_account.id()
+}
+
+/// Get the size of the store file and its WAL file.
+fn get_store_size(dump_file: &Path) -> u64 {
+    let store_file_size = std::fs::metadata(dump_file).unwrap().len();
+    let wal_file_size =
+        std::fs::metadata(format!("{}-wal", dump_file.to_str().unwrap())).unwrap().len();
+    store_file_size + wal_file_size
+}
+
+/// Create a new faucet account.
+fn create_faucet() -> Account {
+    let coin_seed: [u64; 4] = rand::thread_rng().r#gen();
+    let mut rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
+    let key_pair = SecretKey::with_rng(&mut rng);
+    let init_seed = [0_u8; 32];
+
+    let (new_faucet, _seed) = AccountBuilder::new(init_seed)
+        .anchor(AccountIdAnchor::PRE_GENESIS)
+        .account_type(AccountType::FungibleFaucet)
+        .storage_mode(AccountStorageMode::Private)
+        .with_component(RpoFalcon512::new(key_pair.public_key()))
+        .with_component(
+            BasicFungibleFaucet::new(TokenSymbol::new("TEST").unwrap(), 2, Felt::new(u64::MAX))
+                .unwrap(),
+        )
+        .build()
+        .unwrap();
+    new_faucet
+}
+
+fn create_batch(txs: &[ProvenTransaction]) -> ProvenBatch {
+    let account_updates = txs
+        .iter()
+        .map(|tx| (tx.account_id(), BatchAccountUpdate::from_transaction(tx)))
+        .collect();
+    let output_notes = txs.iter().flat_map(|tx| tx.output_notes().iter().cloned()).collect();
+    let input_notes = txs.iter().flat_map(|tx| tx.input_notes().iter().cloned()).collect();
+    ProvenBatch::new_unchecked(
+        BatchId::from_transactions(txs.iter()),
+        Digest::default(),
+        BlockNumber::GENESIS,
+        account_updates,
+        InputNotes::new_unchecked(input_notes),
+        output_notes,
+        BlockNumber::from(u32::MAX),
+    )
+}
+
+fn create_tx(
+    account_id: AccountId,
+    initial_account_hash: Digest,
+    final_account_hash: Digest,
+    output_notes: Vec<Note>,
+    input_notes: Vec<(Note, NoteInclusionProof)>,
+) -> ProvenTransaction {
+    ProvenTransactionBuilder::new(
+        account_id,
+        initial_account_hash,
+        final_account_hash,
+        BlockNumber::from(0),
+        Digest::default(),
```

Comment:
I would try to use the actual block number and block hash. It shouldn't be too much extra work, but would create a more consistent database which should be better for subsequent tests.